lib/engine-universal/src/executable.rs

```diff
@@ -0,0 +1,286 @@
+use std::sync::Arc;
+
+use enumset::EnumSet;
+use loupe::MemoryUsage;
+use rkyv::de::deserializers::SharedDeserializeMap;
+use rkyv::ser::serializers::{
+    AllocScratchError, AllocSerializer, CompositeSerializerError, SharedSerializeMapError,
+};
+use wasmer_compiler::{
+    CompileError, CompileModuleInfo, CompiledFunctionFrameInfo, CpuFeature, CustomSection, Dwarf,
+    Features, FunctionBody, JumpTableOffsets, Relocation, SectionIndex, TrampolinesSection,
+};
+use wasmer_engine::{DeserializeError, Engine};
+use wasmer_types::entity::{EntityRef, PrimaryMap};
+use wasmer_types::{
+    EntityCounts, ExportIndex, FunctionIndex, ImportIndex, LocalFunctionIndex,
+    OwnedDataInitializer, SignatureIndex,
+};
+use wasmer_vm::Artifact;
+
+static MAGIC_HEADER: [u8; 32] =
```

Comment:
TODO: add a comment here that the size matters here so that the data that follows immediately after is naturally aligned. Maybe also a compile time assert?